Components/Queues/Drivers/PhpAmqpLibDriver.php

```diff
@@ -0,0 +1,281 @@
+use PhpAmqpLib\Message\AMQPMessage;
+use PhpAmqpLib\Wire\AMQPTable;
+use Smartbox\Integration\FrameworkBundle\Components\Queues\QueueMessage;
+use Smartbox\Integration\FrameworkBundle\Components\Queues\QueueMessageInterface;
+use Smartbox\Integration\FrameworkBundle\Core\Messages\Context;
+use Smartbox\Integration\FrameworkBundle\DependencyInjection\Traits\UsesSerializer;
+use Smartbox\Integration\FrameworkBundle\Service;
+
+/**
+ * Class PhpAmqpLibDriver.
+ */
+class PhpAmqpLibDriver extends Service implements AsyncQueueDriverInterface
+{
+    use UsesSerializer;
+
+    /**
+     * This field specifies the prefetch window size in octets.
+     * The server will send a message in advance if it is equal to or smaller in size than the available prefetch size
+     * (and also falls into other prefetch limits).
+     * May be set to zero, meaning "no specific limit", although other prefetch limits may still apply.
+     * The prefetch-size is ignored if the no-ack option is set.
+     */
+    const PREFETCH_SIZE = null;
+
+    /**
+     * Represents the amount of message to consume by iteration
+     */
+    const PREFETCH_COUNT = 10;
+
+    /*
+     * Default exchange name
+     */
+    const EXCHANGE_NAME = '';
+
+    /**
+     * Default port to connection
+     */
+    const DEFAULT_PORT = 5672;
+
+    /**
+     * Default host to connect
+     */
+    const DEFAULT_HOST = 'localhost';
+
+    /**
+     * @var string
+     */
+    protected $format = AsyncQueueDriverInterface::FORMAT_JSON;
+
+    /**
+     * @var AMQPChannel|null
+     */
+    private $channel;
+
+    /**
+     * @var array|null
+     */
+    protected $connectionsData;
+
+    /**
+     * @var AMQPStreamConnection
+     */
+    protected $stream;
+
+    public function __destruct()
+    {
+        $this->disconnect();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function configure(string $host, string $username, string $password, string $vhost = null)
+    {
+        $urls = explode(',', $host);
+
+        foreach ($urls as $url) {
+            $parsedUrl = parse_url($url);
+
+            $this->connectionsData[] = [
+                'host' => $parsedUrl['host'] ?? self::DEFAULT_HOST,
+                'user' => $username,
+                'password' => $password,
+                'vhost' => $vhost,
+                'port' => $parsedUrl['port'] ?? self::DEFAULT_PORT,
+            ];
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     * @throws AMQPProtocolException
+     */
+    public function connect()
+    {
+        if (!$this->validateConnection()) {
+            try {
+                $this->stream = AMQPStreamConnection::create_connection($this->connectionsData, []);
+            } catch (AMQPIOException $exception) {
+                throw new AMQPProtocolException($exception->getCode(), $exception->getMessage(), null);
+            }
+        } else if ($this->validateConnection() && !$this->isConnected()) {
```

Comment:
In the else if is not needed to execute again `$this->validateConnection()` because you already did this before in the if statement and the result provoked you are in the else if

Reply:
Fixed.